### PR TITLE
Fixes #19604 vmware.rb vSphere 6.5 GuestOsIdentifier replace

### PR DESCRIPTION
**Pull request by ghost**
Replace the old vSphere 6.0 GuestOsIdentifier with the new vSphere 6.5 GuestOsIdentifier

### CODE REVIEW

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 1]
```diff
@@ -188,140 +188,158 @@ def disk_mode_types
...
+        "rhel4_64Guest" => "Red Hat Enterprise Linux 4 (64-bit)",
+        "rhel4Guest" => "Red Hat Enterprise Linux 4 (32-bit)",
+        "rhel5_64Guest" => "Red Hat Enterprise Linux 5 (64-bit)",
+        "rhel5Guest" => "Red Hat Enterprise Linux 5 (32-bit)",
+        "rhel6_64Guest" => "Red Hat Enterprise Linux 6 (64-bit)",
+        "rhel6Guest" => "Red Hat Enterprise Linux 6 (32-bit)",
+        "rhel7_64Guest" => "Red Hat Enterprise Linux 7 (64-bit)",
+        "rhel7Guest" => "Red Hat Enterprise Linux 7 (32-bit)",
+        "sjdsGuest" => "Sun Java Desktop System",
+        "sles10_64Guest	" => "Suse Linux Enterprise Server 10 (64-bit)",
```
Comment (mmoll) [inline app/models/compute_resources/foreman/model/vmware.rb]: whitespace in the key

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 2]
```diff
@@ -188,140 +188,158 @@ def disk_mode_types
...
+        "sles10Guest" => "Suse linux Enterprise Server 10 (32-bit)",
```
Comment (mmoll) [inline app/models/compute_resources/foreman/model/vmware.rb]: linux -> Linux

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 3]
```diff
@@ -188,140 +188,158 @@ def disk_mode_types
...
+        "sles11_64Guest" => "Suse Linux Enterprise Server 11 (64-bit)",
+        "sles11Guest" => "Suse linux Enterprise Server 11 (32-bit)",
```
Comment (mmoll) [inline app/models/compute_resources/foreman/model/vmware.rb]: linux -> Linux

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 4]
```diff
@@ -188,140 +188,158 @@ def disk_mode_types
...
+        "sles12_64Guest" => "Suse Linux Enterprise Server 12 (64-bit)",
+        "sles12Guest" => "Suse linux Enterprise Server 12 (32-bit)",
```
Comment (mmoll) [inline app/models/compute_resources/foreman/model/vmware.rb]: linux -> Linux

**Review by mmoll — CHANGES_REQUESTED** [review]

**Comment by mmoll** [comment]
👍 for sorting, this will ease diffs in the future

**Comment by mmoll** [comment]
while you're at it, could you add `vmx-13` in `vm_hw_versions`, per https://kb.vmware.com/selfservice/microsites/search.do?language=en_US&cmd=displayKC&externalId=1003746 ?

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
looks correct, squashing the commits together and changing the commit message can be done while merging

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 5]
```diff
@@ -188,140 +188,158 @@ def disk_mode_types
...
-        "asianux4Guest" => "Asianux Server 4 (32-bit)",
-        "asianux4_64Guest" => "Asianux Server 4 (64-bit)",
-        "asianux5_64Guest" => "Asianux Server 5 (64-bit)",
-        "opensuseGuest" => "SUSE OpenSUSE (32-bit)",
-        "opensuse64Guest" => "SUSE OpenSUSE Linux (64-bit)",
-        "fedoraGuest" => "Fedora Linux",
+        "debian8Guest" => "Debian GNU/Linux 8 (32-bit)",
+        "debian9_64Guest" => "Debian GNU/Linux 9 (64-bit)",
+        "debian9Guest" => "Debian GNU/Linux 9 (32-bit)",
+        "dosGuest" => "MS-DOS.",
```
Comment (timogoebel) [inline app/models/compute_resources/foreman/model/vmware.rb]: "dosGuest" => "Microsoft MS-DOS",

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 6]
```diff
@@ -188,140 +188,158 @@ def disk_mode_types
...
         "openServer6Guest" => "SCO OpenServer 6",
+        "opensuse64Guest" => "OpenSUSE Linux (64-bit)",
+        "opensuseGuest" => "OpenSUSE Linux (32-bit)",
+        "oracleLinux6_64Guest" => "Oracle 6 (64-bit)",
+        "oracleLinux64Guest" => "Oracle Linux 4/5 (64-bit)",
+        "oracleLinux6Guest" => "Oracle 6 (32-bit)",
+        "oracleLinux7_64Guest" => "Oracle 7 (64-bit)",
+        "oracleLinux7Guest" => "Oracle 7 (32-bit)",
+        "oracleLinuxGuest" => "Oracle Linux 4/5",
+        "os2Guest" => "OS/2",
```
Comment (timogoebel) [inline app/models/compute_resources/foreman/model/vmware.rb]: => "IBM OS/2"

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 7]
```diff
@@ -188,140 +188,158 @@ def disk_mode_types
...
+        "vmwarePhoton64Guest" => "VMware Photon (64-bit)",
+        "win2000AdvServGuest" => "Windows 2000 Advanced Server",
+        "win2000ProGuest" => "Windows 2000 Professional",
+        "win2000ServGuest" => "Windows 2000 Server",
+        "win31Guest" => "Windows 3.1",
+        "win95Guest" => "Windows 95",
+        "win98Guest" => "Windows 98",
+        "windows7_64Guest" => "Windows 7 (64-bit)",
+        "windows7Guest" => "Windows 7 (32-bit)",
+        "windows7Server64Guest" => "Windows Server 2008 R2 (64-bit)",
```
Comment (timogoebel) [inline app/models/compute_resources/foreman/model/vmware.rb]: "Microsoft Windows Server 2008"

... same for the other MS-Products.

**Comment by timogoebel** [comment]
@nokrite : Thanks a lot for doing this. I believe some descriptions need some manual tuning. I left some comments inline.

**Review by timogoebel — CHANGES_REQUESTED** [review]

File: app/models/compute_resources/foreman/model/vmware.rb (hunk 7 above)
Comment (nokrite) [inline app/models/compute_resources/foreman/model/vmware.rb]: Cutting "R2" and "(64-Bit)" ok - but I didnt see sth. compareable in the other MS-Products. Can you specify this that we dont talk at cross-purposes ? Thanks in advance.

**Comment by dLobatog** [comment]
[test foreman]
[test katello]
[test upgrade]

This triggers automated tests in our CI for people who are not yet members of @theforeman organization. Thanks for your PR!

**Review by timogoebel — APPROVED** [review]

**Comment by timogoebel** [comment]
Closing as the user has deleted his github account.

**Pull request closed without merging**